apollo-router/src/spec/query/change.rs

```diff
@@ -73,7 +112,23 @@ impl<'a> QueryHashVisitor<'a> {
                 ">=0.1.0",
                 JOIN_TYPE_DIRECTIVE_NAME,
             ),
+        };
+
+        visitor.hash_schema()?;
+
+        Ok(visitor)
+    }
+
+    pub(crate) fn hash_schema(&mut self) -> Result<(), BoxError> {
```

Comment:
As an idea for follow up PR: What if use struct destructuring (without `...`) for accessing fields on `schema`? 
In that way, we can ensure that all fields are considered for hashing, so if new fields are added to `apollo-rs`, they will result in compiler errors.
That way, we protect against stuff that would be added to the GraphQL spec in the future, e.g., support for directives in a new location.

Reply:
it could be useful for the other structures we use elsewhere in the visitor. Here, the schema also holds a list of types, but the point of this algorithm would be to ignore irrelevant types